pelican/plugins/photos/photos.py

```diff
@@ -2103,85 +2123,109 @@ def replace_inline_images(content, inline_images):
+            if what == "photo":
+                content._content = content._content.replace(
+                    image_string,
+                    "".join(
+                        (
+                            "<",
+                            m.group("tag"),
+                            m.group("attrs_before"),
+                            m.group("src"),
+                            "=",
+                            m.group("quote"),
+                            "{siteurl}/{filename}".format(
+                                siteurl=siteurl,
+                                filename=image.image.web_filename,
+                            ),
+                            m.group("quote"),
+                            extra_attributes,
+                            m.group("attrs_after"),
+                        )
+                    ),
                 )
-            )
-
-            if image.caption:
+            elif what == "lightbox" and tag == "img":
+                lightbox_attr_list = [""]
+                gallery_name = value.split("/")[0]
                 lightbox_attr_list.append(
                     '{}="{}"'.format(
-                        pelican_settings["PHOTO_LIGHTBOX_CAPTION_ATTR"],
-                        str(image.caption),
+                        pelican_settings["PHOTO_LIGHTBOX_GALLERY_ATTR"], gallery_name
                     )
                 )
-
-            lightbox_attrs = " ".join(lightbox_attr_list)
-
-            content._content = content._content.replace(
-                image_string,
-                "".join(
-                    (
-                        "<a href=",
-                        m.group("quote"),
-                        "{siteurl}/{filename}".format(
-                            siteurl=pelican_settings["SITEURL"],
-                            filename=image.image.web_filename,
-                        ),
-                        m.group("quote"),
-                        lightbox_attrs,
-                        "><img",
-                        m.group("attrs_before"),
-                        "src=",
-                        m.group("quote"),
-                        "{siteurl}/{filename}".format(
-                            siteurl=pelican_settings["SITEURL"],
-                            filename=image.thumb.web_filename,
-                        ),
-                        m.group("quote"),
-                        extra_attributes,
-                        m.group("attrs_after"),
-                        "</a>",
+                if image.caption:
+                    lightbox_attr_list.append(
+                        '{}="{}"'.format(
+                            pelican_settings["PHOTO_LIGHTBOX_CAPTION_ATTR"],
+                            str(image.caption),
+                        )
                     )
-                ),
+                lightbox_attrs = " ".join(lightbox_attr_list)
+                content._content = content._content.replace(
```

Comment:
I gotta wonder... would it make sense to define these as template strings in the package somewhere, and then just have one path through here (if a template is set, load it, otherwise load the default template, and then render them both)?